SimG4CMS/HGCalTestBeam/plugins/HGCPassive.cc

```diff
@@ -162,8 +185,30 @@ std::map<G4LogicalVolume*,std::pair<unsigned int,std::string>>::iterator HGCPass
       }
     }
   }
+  if (topLV_ == 0) {
+    if (std::string(plv->GetName()) == motherName_) topLV_ = plv;
+  }
   return itr;
 }
 
+void HGCPassive::storeInfo(const HGCPassive::volumeIterator it,
+			   G4LogicalVolume* plv, unsigned int copy, 
+			   double time, double energy) {
+
+  std::pair<G4LogicalVolume*,unsigned int> key(plv,copy);
+  auto itr = store_.find(key);
+  if (itr == store_.end()) {
+    store_[key] = std::pair<double,double>(time,0.0);
```

Comment:
HI @bsunanda - please avoid the second find by setting the second item in the pair to energy instead of 0. (and then add an else just below)